uvdat/core/rest/data.py

```diff
@@ -60,20 +57,42 @@
         core_vectorfeature t,
         bounds
     WHERE
-        t.map_layer_id = %(map_layer_id)s
+        t.vector_data_id = %(vector_data_id)s
         AND ST_Intersects(
             ST_Transform(t.geometry, %(srid)s),
             ST_Transform(bounds.geom, %(srid)s)
         )
+        REPLACE_WITH_FILTERS
 )
 SELECT ST_AsMVT(mvtgeom.*) FROM mvtgeom
 ;
 """
 
 
-class RasterMapLayerViewSet(GenericViewSet, mixins.RetrieveModelMixin, LargeImageFileDetailMixin):
-    queryset = RasterMapLayer.objects.select_related('dataset').all()
-    serializer_class = RasterMapLayerSerializer
+def get_filter_string(filters=None, string=None, key_prefix=None):
+    if string is None:
+        string = ''
+    if key_prefix is None:
+        key_prefix = ''
+    else:
+        key_prefix += ','
+    if filters is not None:
+        for key, value in filters.items():
+            try:
+                value = ast.literal_eval(value)
+            except ValueError:
+                pass
+            if isinstance(value, dict):
+                string = get_filter_string(value, string, key_prefix + key)
+            else:
+                key_path = '{' + key_prefix + key + '}'
+                string += f"\n\t\tAND t.properties #>> '{key_path}' = '{value}'"
+    return string
```

Comment:
I think this is too complicated and presents a maintainability concern. I think it's fair to define the following constraints:

1. Nested keys must be specified with javascript syntax (e.g `a.b.c`).
2. Filter key/value pairs must be a primitive type. Objects are not supported.
3. To distinguish between strings and other primitive types, string values must be quoted in strings when passed to the backend (e.g. `"1"` vs `1`).

Given those constraints, what do you think about this simplified version:

```suggestion
def get_filter_string(filters: dict | None = None):
    if filters is None:
        return ''

    return_str = ''
    for key, value in filters.items():
        key_path = key.replace('.', ',')
        return_str += f" AND t.properties #>> '{{{key_path}}}' = {value}"
    return return_str

```

To be honest even this is slight overkill for what we are planning to support at the moment, but it will at least allow basic filtering and nested property filtering.

Reply:
I think we have to include the single quotes around the value. When I use this function and test the vector tile endpoint with a query of `frame=0`, the following error is raised:
```
django.db.utils.ProgrammingError: operator does not exist: text = integer                                   
LINE 50:          AND t.properties #>> '{frame}'  = 0                                                                                                               
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.
```

Reply:
Ah true, the `#>>` operator itself returns a `text` field. I guess for now we'll have to just use strings. In that case, I've updated the above suggestion. It does make the implementation easier, at least.

Reply:
Ok great, I'll commit this suggestion.

Reply:
I've updated the `jupyter-explorer` branch to be compatible with these changes: [52352cf](https://github.com/OpenGeoscience/uvdat/pull/96/commits/52352cf9c71a559d05d8d8e86830514a8247bd84).